src/DynamoCore/Models/WorkspaceModel.cs

```diff
@@ -1075,11 +1075,33 @@ private void GenerateCombinedGraph(bool isGroupLayout)
 
             foreach (NoteModel note in Notes)
             {
-                // Link a note to the nearest node
-                GraphLayout.Node nd = combinedGraph.Nodes.OrderBy(node =>
-                    Math.Pow(node.X + node.Width / 2 - note.X - note.Width / 2, 2) +
-                    Math.Pow(node.Y + node.Height / 2 - note.Y - note.Height / 2, 2)).FirstOrDefault();
+                AnnotationModel group = Annotations.Where(
+                    g => g.SelectedModels.Contains(note)).ToList().FirstOrDefault();
 
+                GraphLayout.Node nd = null;
+
+                if (!isGroupLayout || group == null)
+                {
+                    // If note is not part of a group, link to the nearest node in the graph
+                    nd = combinedGraph.Nodes.OrderBy(node =>
```

Comment:
Two different statements are needed here:

- The one in the `if` block can search directly for `GraphLayout.Node` object in the GraphLayout layer.

- The one in the `else` block needs to search for `NodeModel` object inside the particular `AnnotationModel`,
since an `AnnotationModel` is treated as one `GraphLayout.Node` object in the GraphLayout layer.